### PR TITLE
use synset for comparison

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual.
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Regression tests have to complete successfully.
- [ ] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test.
- [ ] 6. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.

### CODE REVIEW

**Comment by wangcj05** [comment]
@joshua-cogliati-inl @mandd FYI. This PR contains the changes that we can accelerate the similarity check.

File: src/nlp/similarity/speedup.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+import nltk
+from nltk import word_tokenize as tokenizer
+from nltk.corpus import wordnet as wn
+
+
+def convertToSynsets(wordSet):
+  # keep the order (works for python3.7+)
+  wordList = list(dict.fromkeys(wordSet))
+  synsets = [list(wn.synsets(word)) for word in wordList]
+  return list(wordList), synsets
```
Comment (mandd) [inline src/nlp/similarity/speedup.py]: wordList is already a list no?

File: src/nlp/similarity/speedup.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+      bestSyn = syn
+      max = temp
+  return bestSyn
+
+def synsetListSimilarity(synsetList1, synsetList2, delta=0.85):
+  """
+
+  """
+  similarity = delta * semanticSimilaritySynsetList(synsetList1, synsetList2) + (1.0-delta)* wordOrderSimilaritySynsetList(synsetList1, synsetList2)
+  return similarity
```
Comment (mandd) [inline src/nlp/similarity/speedup.py]: it will be interesting to play with delta

**Comment by mandd** [comment]
I started to take a look at it and i left few comments, I'll let Josh test is on the outage dataset.
We have few PRs that likely need to be merged.
Let's talk on Monday on how to proceed

**Comment by wangcj05** [comment]
@mandd This PR is ready to be reviewed.

**Pull request merged**